mantidproject/mantid
# Fixing cppcheck warnings

Starting with what was pointed out by [cppcheck on master](https://builds.mantidproject.org/job/pull_requests-cppcheck/47481/cppcheckResult/) and fixing mostly pass-by-reference issues. This is not intended to get everything.

**To test:**

Look at the cppcheck build and see that there are less than 361 issues.

*There is no associated issue.*

*This does not require release notes* because it is fixing issues from static analysis tools.

**Once this is merged the number of allowed warnings on builds should be reduced**

The version into `ornl-next` is #32098.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
